### ISSUE TITLE
Performance degradation for cpc.get_property('se-version')

### ISSUE THREAD

**Issue by andy-maier** [zhmcclient/python-zhmcclient#1418]
The zhmc_lpar_list Ansible module got slower with zhmcclient to 1.13.1 compared to 1.12.4. The test used ibm_zhmc collection version 1.7.4.

It seems that zhmcclient is taking a long time to return the se-version here: https://github.com/zhmcclient/zhmc-ansible-modules/blob/master/plugins/modules/zhmc_lpar_list.py#L360
```
se_version = parent_cpc.get_property('se-version')
```
Measuring the time for just that function results in:
* zhmcclient 1.13.1: around 20 s
* zhmcclient 1.12.4: below 1 ms

The issue for the zhmc_lpar_list Ansible module is that this time difference gets multiplied by the number of LPARs returned.

This ticket is to investigate what caused the change and to improve the performance, if possible.

**Comment by andy-maier** [comment]
Using a simple test playbook that performs zhmc_list, I could reproduce the performance difference (using ibm_zhmc 1.8.2-dev1 from the master branch):

with zhmcclient 1.13.1:
```
2024-01-25T08:27:26+0100 DEBUG zhmc_lpar_list 38547 Module entry: params: {'hmc_host': '9.152.150.166', 'log_file': 'lpar_list.log', 'additional_properties': []
2024-01-25T08:27:26+0100 DEBUG zhmcclient.hmc 38547 Using the only HMC specified without verifying its availability: 9.152.150.166
2024-01-25T08:27:26+0100 DEBUG zhmcclient.hmc 38547 Request: GET https://9.152.150.166:6794/api/version, headers: {'User-Agent': 'python-zhmcclient/1.13.1', 'Co
2024-01-25T08:27:27+0100 DEBUG zhmcclient.hmc 38547 Respons: GET https://9.152.150.166:6794/api/version, status: 200, headers: {'Strict-Transport-Security': 'ma
2024-01-25T08:27:27+0100 DEBUG zhmc_lpar_list 38547 Listing LPARs of all managed CPCs (List permitted LPARs with default properties)
2024-01-25T08:27:27+0100 DEBUG zhmcclient.hmc 38547 Using the only HMC specified without verifying its availability: 9.152.150.166
2024-01-25T08:27:27+0100 DEBUG zhmcclient.hmc 38547 Request: POST https://9.152.150.166:6794/api/sessions, headers: {'User-Agent': 'python-zhmcclient/1.13.1', '
2024-01-25T08:27:28+0100 DEBUG zhmcclient.hmc 38547 Respons: POST https://9.152.150.166:6794/api/sessions, status: 200, headers: {'Strict-Transport-Security': '
2024-01-25T08:27:28+0100 DEBUG zhmcclient.hmc 38547 Request: GET https://9.152.150.166:6794/api/console/operations/list-permitted-logical-partitions (console <u
2024-01-25T08:27:29+0100 DEBUG zhmcclient.hmc 38547 Respons: GET https://9.152.150.166:6794/api/console/operations/list-permitted-logical-partitions (console <u
2024-01-25T08:27:29+0100 DEBUG zhmcclient.hmc 38547 Request: GET https://9.152.150.166:6794/api/cpcs, headers: {'User-Agent': 'python-zhmcclient/1.13.1', 'Conte
2024-01-25T08:27:29+0100 DEBUG zhmcclient.hmc 38547 Respons: GET https://9.152.150.166:6794/api/cpcs, status: 200, headers: {'Strict-Transport-Security': 'max-a
2024-01-25T08:27:29+0100 DEBUG zhmcclient.hmc 38547 Request: GET https://9.152.150.166:6794/api/cpcs/4f01576a-c3f6-3224-a951-b1bf361886a4 (cpc P0000M96), header
2024-01-25T08:27:46+0100 DEBUG zhmcclient.hmc 38547 Respons: GET https://9.152.150.166:6794/api/cpcs/4f01576a-c3f6-3224-a951-b1bf361886a4 (cpc P0000M96), status
2024-01-25T08:27:46+0100 DEBUG zhmcclient.hmc 38547 Request: GET https://9.152.150.166:6794/api/cpcs/4f01576a-c3f6-3224-a951-b1bf361886a4 (cpc P0000M96), header
2024-01-25T08:28:03+0100 DEBUG zhmcclient.hmc 38547 Respons: GET https://9.152.150.166:6794/api/cpcs/4f01576a-c3f6-3224-a951-b1bf361886a4 (cpc P0000M96), status
2024-01-25T08:28:03+0100 DEBUG zhmcclient.hmc 38547 Request: GET https://9.152.150.166:6794/api/cpcs/4f01576a-c3f6-3224-a951-b1bf361886a4 (cpc P0000M96), header
2024-01-25T08:28:20+0100 DEBUG zhmcclient.hmc 38547 Respons: GET https://9.152.150.166:6794/api/cpcs/4f01576a-c3f6-3224-a951-b1bf361886a4 (cpc P0000M96), status
2024-01-25T08:28:20+0100 DEBUG zhmcclient.hmc 38547 Request: GET https://9.152.150.166:6794/api/cpcs/4f01576a-c3f6-3224-a951-b1bf361886a4 (cpc P0000M96), header
2024-01-25T08:28:37+0100 DEBUG zhmcclient.hmc 38547 Respons: GET https://9.152.150.166:6794/api/cpcs/4f01576a-c3f6-3224-a951-b1bf361886a4 (cpc P0000M96), status
2024-01-25T08:28:37+0100 DEBUG zhmcclient.hmc 38547 Request: GET https://9.152.150.166:6794/api/cpcs/4f01576a-c3f6-3224-a951-b1bf361886a4 (cpc P0000M96), header
2024-01-25T08:28:54+0100 DEBUG zhmcclient.hmc 38547 Respons: GET https://9.152.150.166:6794/api/cpcs/4f01576a-c3f6-3224-a951-b1bf361886a4 (cpc P0000M96), status
2024-01-25T08:28:54+0100 DEBUG zhmcclient.hmc 38547 Request: GET https://9.152.150.166:6794/api/cpcs/4f01576a-c3f6-3224-a951-b1bf361886a4 (cpc P0000M96), header
... stopped here with Ctrl-C
2024-01-25T08:29:07+0100 DEBUG zhmcclient.hmc 38547 Request: DELETE https://9.152.150.166:6794/api/sessions/this-session, headers: {'User-Agent': 'python-zhmccl
2024-01-25T08:29:08+0100 DEBUG zhmcclient.hmc 38547 Respons: DELETE https://9.152.150.166:6794/api/sessions/this-session, status: 204, headers: {'Strict-Transpo
```

with zhmcclient 1.12.4:

```
2024-01-25T08:30:10+0100 DEBUG zhmc_lpar_list 38605 Module entry: params: {'hmc_host': '9.152.150.166', 'log_file': 'lpar_list.log', 'additional_properties': []
2024-01-25T08:30:10+0100 DEBUG zhmcclient.hmc 38605 Request: GET https://9.152.150.166:6794/api/version, headers: {'User-Agent': 'python-zhmcclient/1.12.4', 'Co
2024-01-25T08:30:11+0100 DEBUG zhmcclient.hmc 38605 Respons: GET https://9.152.150.166:6794/api/version, status: 200, headers: {'Strict-Transport-Security': 'ma
2024-01-25T08:30:11+0100 DEBUG zhmc_lpar_list 38605 Listing LPARs of all managed CPCs (List permitted LPARs with default properties)
2024-01-25T08:30:11+0100 DEBUG zhmcclient.hmc 38605 Request: POST https://9.152.150.166:6794/api/sessions, headers: {'User-Agent': 'python-zhmcclient/1.12.4', '
2024-01-25T08:30:12+0100 DEBUG zhmcclient.hmc 38605 Respons: POST https://9.152.150.166:6794/api/sessions, status: 200, headers: {'Strict-Transport-Security': '
2024-01-25T08:30:12+0100 DEBUG zhmcclient.hmc 38605 Request: GET https://9.152.150.166:6794/api/console/operations/list-permitted-logical-partitions (console <u
2024-01-25T08:30:13+0100 DEBUG zhmcclient.hmc 38605 Respons: GET https://9.152.150.166:6794/api/console/operations/list-permitted-logical-partitions (console <u
2024-01-25T08:30:13+0100 DEBUG zhmcclient.hmc 38605 Request: DELETE https://9.152.150.166:6794/api/sessions/this-session, headers: {'User-Agent': 'python-zhmccl
2024-01-25T08:30:13+0100 DEBUG zhmcclient.hmc 38605 Respons: DELETE https://9.152.150.166:6794/api/sessions/this-session, status: 204, headers: {'Strict-Transpo
2024-01-25T08:30:13+0100 DEBUG zhmc_lpar_list 38605 Module exit (success): changed: False, lpars: [{'cpc_name': 'P0000M96', 'se_version': '2.14.1', 'parent': '/
```

**Comment by andy-maier** [comment]
I found the problem and fixed it in PR https://github.com/zhmcclient/python-zhmcclient/pull/1419. There will be another fix PR in the ibm_zhmc Ansible collection, but the performance improvement comes from this PR.

See the commit message of the PR for details on what happened and how it was fixed.

The PR will be released in 1.14.0 and will be rolled back to 1.13.x.
